packages/react-admin-core/src/table/Pagination.tsx

```diff
@@ -1,17 +1,90 @@
 import { Grid, IconButton, Toolbar, Typography } from "@material-ui/core";
 import TableCell from "@material-ui/core/TableCell";
+import { FirstPage, LastPage } from "@material-ui/icons";
 import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft";
 import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight";
+import { Field } from "@vivid-planet/react-admin-form";
 import * as React from "react";
+import { Form } from "react-final-form";
+import * as sc from "./Pagination.sc";
 import { IPagingInfo } from "./paging/IPagingInfo";
 
-interface IProps {
+interface IPaginationProps {
     totalCount: number;
     pagingInfo: IPagingInfo;
     rowName?: string | ((count: number) => string);
 }
 
-export const TablePagination: React.FunctionComponent<IProps> = ({ totalCount, pagingInfo, rowName }) => {
+interface IInputProps {
+    pagingInfo: IPagingInfo;
+    lastInteraction?: "button" | "input";
+    setLastInteraction: React.Dispatch<React.SetStateAction<"button" | "input" | undefined>>;
+}
+
+export const PageInput: React.FC<IInputProps> = ({
+    pagingInfo: { fetchSpecificPage, totalPages, currentPage },
+    lastInteraction,
+    setLastInteraction,
+}) => {
+    const validation = (value: number, totalPages: number | null | undefined) => {
+        if (totalPages && value >= 1 && value <= totalPages) {
+            return true;
+        }
+        return false;
+    };
+
+    const onSubmit = async (values: any) => {
+        fetchSpecificPage && fetchSpecificPage(values.page);
```

Comment:
mach ganz oben in der component if (!fetchSpecificPage) return null; Oder, noch besser mach es pflicht in der props dann brauchst du da nicht prüfen